src/urllib3/connection.py

```diff
@@ -500,6 +515,11 @@ def connect(self) -> None:
                 SystemTimeWarning,
             )
 
+        if self._connecting_to_proxy and self.proxy_config:
```

Comment:
@sethmlarson this might not work.

``self._connecting_to_proxy`` will remain True when we're using the absolute URI forwarding mode after the block in L486. I suspect that's why @avi364 is relying on it. I don't like relying on it though, seems brittle.

Could we check ``not self._is_using_tunnel() and self.proxy_config`` instead? That will default to the absolute UI forwarding mode. It might be better to check if ``proxy_ssl_context`` is set as well.

Reply:
Makes complete sense to me! @avi364 could you implement these changes and then we can merge?